msbuild/Xamarin.iOS.Tasks.Core/Tasks/CompileAppManifestTaskBase.cs

```diff
@@ -236,6 +246,84 @@ bool Compile (PDictionary plist)
 			return !Log.HasLoggedErrors;
 		}
 
+		void SetRequiredArchitectures (PDictionary plist)
+		{
+			PObject capabilities;
+
+			if (plist.TryGetValue (ManifestKeys.UIRequiredDeviceCapabilities, out capabilities)) {
+				if (capabilities is PArray) {
+					var architectureValues = new HashSet<string> (new[] { "armv6", "armv7", "arm64" });
+					var array = (PArray) capabilities;
+
+					// Remove any architecture values
+					for (int i = 0; i < array.Count; i++) {
+						var value = array[i] as PString;
+
+						if (value == null || !architectureValues.Contains (value.Value))
+							continue;
+
+						array.RemoveAt (i);
+					}
+
+					// If-and-only-if the TargetArchitecture is a single architecture, set it as a required device capability
+					switch (architectures) {
+					case TargetArchitecture.ARM64:
+						array.Add (new PString ("arm64"));
+						break;
+					case TargetArchitecture.ARMv7:
```

Comment:
According to my reading of the docs, you only use "arm64" if it *requires* arm64, i.e. it won't run on armv7.